### DISCUSSION TITLE
Dienste hinzufügen: Service-Reihenfolge beim Ausdruck: Build 5.22.4401

### DISCUSSION THREAD

**Discussion by cmairamhof** [e-netze/webgis-community#7] [General]
Wird unter "Dienste hinzufügen" ein Hintergrunddienst z.B. Orthofoto hinzugefügt, stellt der Viewer die Reihenfolge der Dienste, z.B. DKM über Orthofoto richtig dar.

Beim Ausdruck bleibt diese Reihenfolge unberücksichtigt, das neu hinzugefügte Orthofoto deckt dann die DKM ab. Dieses Verhalten tritt auch schon bei älteren Builds, z.B.  5.22.1807 auf.

**Comment by gis-eni** [comment]
Dienste bekommen beim Einfügen immer die letzte Ordnungsnummer. Dh sie werden ganz oben eingefügt. Bei Basemaps/Tiling Diensten ist es aber so, dass die am Client (durch Leaflet) in einen eigenen Tiles Container kommen in der Regel immer unter dem Overlay Container liegen (im Overlay Container liegen die "normal" Vektordaten Dienste).

Darum schaut es am Client einmal wie gewünscht aus. Die Ordungsnummer wird durch die beiden Container quasi over-ruled.

Wenn man jetzt druckt, erfolgt die Kartenerstellung am Server. Dort gibt es die Leaflet-Container nicht und die Dienste werden jetzt tatsächlich nach der Ordnungsnummer sortiert gezeichnet. Der letzte Dienst, der einfügt wurde (bei dir der Tiling Dienst) wird jetzt als letzter über alles anderen gezeichnet und verdeckt alles.

Dieses Verhalten hab ich jetzt einmal als Bug deklariert. Ab Build 5.22.4602 wird es dann so sein, dass beim hinzuladen der Dienste überprüft wird, ob es sich um einen Basemap Layer handelt. Wenn ja, bekommt er einn Ordnungsnummer unter den Vektor Daten. Damit wird er auch beim Drucken "unten" gezeichnet.

**Reply by cmairamhof** [reply]
Wir wollten das gerade im Build 5.22.4902 testen, vgl. https://gis.ktn.gv.at/webgisviewer/atlas-mobile/map/Basiskarten/0rientierung%20u.%20Kataster

Ja, der Kataster ist nun "über" dem neu hinzugefügten Hintergrund. Ein weiterer Service-Layer, die Verwaltungsgrenzen, bleiben weiter hinter dem Hintergrund versteckt. Kannst du das bitte nochmal ansehen?

**Comment by gis-eni** [comment]
Ich bin gerade bei dieser Karte eingestiegen und hab alle möglichen Hintergrunddienste dazugeladen. Die Verwaltungsgrenzen bleiben dabei immer über den Hintergrunddiensten:

![image](https://user-images.githubusercontent.com/62870568/207067339-c6be6667-4732-4c3f-87a8-4ead9cc53319.png)

Passt meiner Meinung nach.
